consensus/src/quorum_store/utils.rs

```diff
@@ -412,7 +412,11 @@ impl ProofQueue {
                             cur_unique_txns
                                 + txn_summaries
                                     .iter()
-                                    .filter(|txn_summary| !filtered_txns.contains(txn_summary))
+                                    .filter(|txn_summary| {
+                                        !filtered_txns.contains(txn_summary)
+                                            && txn_summary.expiration_timestamp_secs
```

Comment:
Isn't `latest_block_timestamp` in usecs and `expiration_timestamp_secs` in secs?

Reply:
Also, don't you need to apply the same filter below when actually updating cur_unique_txns? Better yet, we should only apply the filter once?

Reply:
Thanks a lot for noticing this. Are you sure the `expiration_timestamp_secs` is not counted from the UNIX timestamp? I was searching for the usage of `expiration_timestamp_secs` in the code and found some usages like [this]
(https://github.com/aptos-labs/aptos-core/blob/main/api/goldens/aptos_api__tests__transactions_test__test_post_bcs_format_transaction.json#L7) where the `expiration_timestamp_secs` has been instantiated with a very large value.

 If `expiration_timestamp_secs` is not from the UNIX timestamp, then when is it from?

Reply:
It's counted from the UNIX epoch time, but the transaction expiration time is in `secs` and the block timestamp is in `usecs` so we have to convert units.

The examples are probably just examples. Transaction emitter is using seconds here https://github.com/aptos-labs/aptos-core/blob/9c2f7d32eed87842d8c6229c8a4700e1acf67ae1/sdk/src/transaction_builder.rs#L254

If you're unsure, maybe you can print some debug logs and run forge. In general, we should run some tests / write unit tests to ensure the expiration is working correctly.